From 6bd9a59c53eeafc53b3c69419366276ae52d8be0 Mon Sep 17 00:00:00 2001
From: David Gow <david@ingeniumdigital.com>
Date: Fri, 3 Mar 2023 17:31:21 +0800
Subject: [PATCH] sd_opl2alsa: Don't include x86-specific headers

As part of an old direct IO port access feature which I gave up on long
ago, id_sd_opl2alsa.c included <sys/io.h>.

This header only exists on x86 platforms, so Omnispeak wouldn't build
on, e.g., aarch64 if WITH_ALSA was declared. It's pretty unlikely that
a non-x86 machine has an OPL3 compatible soundcard, but that's no reason
not to support it if it does.

Of course, all of this should go away and be replaced with liboplhw, but
for now, let's fix it.

Thanks to Martin Hauke for creating openSUSE packages, whose failed
aarch64 build uncovered this issue:
https://build.opensuse.org/package/live_build_log/home:mnhauke:games/omnispeak/openSUSE_Tumbleweed/aarch64
---
 src/id_sd_opl2alsa.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/id_sd_opl2alsa.c b/src/id_sd_opl2alsa.c
index 433eb49..f94fe20 100644
--- a/src/id_sd_opl2alsa.c
+++ b/src/id_sd_opl2alsa.c
@@ -21,7 +21,6 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #include <stdint.h>
 #include <stdio.h>
 #include <string.h>
-#include <sys/io.h>
 
 #include <alsa/asoundlib.h>
 #include <sound/asound_fm.h>
-- 
2.40.0

